sklearn/notebooks/hp_tuning_auto_mpg/ml_engine_hp_tuning_training.ipynb

```diff
@@ -0,0 +1,907 @@
+    "# Part 0: Setup\n",
+    "* [Create a project on GCP](https://cloud.google.com/resource-manager/docs/creating-managing-projects)\n",
+    "* [Create a Google Cloud Storage Bucket](https://cloud.google.com/storage/docs/quickstart-console)\n",
+    "* [Enable Cloud Machine Learning Engine and Compute Engine APIs](https://console.cloud.google.com/flows/enableapi?apiid=ml.googleapis.com,compute_component&_ga=2.217405014.1312742076.1516128282-1417583630.1516128282)\n",
+    "* [Install Cloud SDK](https://cloud.google.com/sdk/downloads)\n",
+    "* [Install scikit-learn](http://scikit-learn.org/stable/install.html) [Optional: used if running locally]\n",
+    "* [Install pandas](https://pandas.pydata.org/pandas-docs/stable/install.html) [Optional: used if running locally]\n",
+    "* [Install cloudml-hypertune](https://pypi.org/project/cloudml-hypertune/) [Optional: used if running locally]\n",
+    "\n",
+    "These variables will be needed for the following steps.\n",
+    "* `TRAINER_PACKAGE_PATH <./auto_mpg_hp_tuning>` - A packaged training application that will be staged in a Google Cloud Storage location. The model file created below is placed inside this package path.\n",
+    "* `MAIN_TRAINER_MODULE <auto_mpg_hp_tuning.train>` - Tells ML Engine which file to execute. This is formatted as follows <folder_name.python_file_name>\n",
+    "* `JOB_DIR <gs://$BUCKET_ID/scikit_learn_job_dir>` - The path to a Google Cloud Storage location to use for job output.\n",
+    "* `RUNTIME_VERSION <1.9>` - The version of Cloud ML Engine to use for the job. If you don't specify a runtime version, the training service uses the default Cloud ML Engine runtime version 1.0. [See the list of runtime versions for more information](https://cloud.google.com/ml-engine/docs/tensorflow/runtime-version-list).\n",
+    "* `PYTHON_VERSION <3.5>` - The Python version to use for the job. Python 3.5 is available with runtime version 1.4 or greater. If you don't specify a Python version, the training service uses Python 2.7.\n",
+    "* `HPTUNING_CONFIG <hptuning_config.yaml>` - Path to the job configuration file.\n",
+    "\n",
+    "** Replace: **\n",
+    "* `PROJECT_ID <YOUR_PROJECT_ID>` - with your project's id. Use the PROJECT_ID that matches your Google Cloud Platform project.\n",
+    "* `BUCKET_ID <YOUR_BUCKET_ID>` - with the bucket id you created above.\n",
+    "* `JOB_DIR <gs://YOUR_BUCKET_ID/scikit_learn_job_dir>` - with the bucket id you created above.\n",
+    "* `REGION <REGION>` - select a region from [here](https://cloud.google.com/ml-engine/docs/regions) or use the default '`us-central1`'. The region is where the model will be deployed."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "env: PROJECT_ID=PROJECT_ID\n",
+      "env: BUCKET_ID=BUCKET_ID\n",
+      "env: JOB_DIR=gs://BUCKET_ID/scikit_learn_job_dir\n",
```

Comment:
This line did not work for me. The system did not replace BUCKET_ID with its value. Did it work for you?

Reply:
Oh you are supposed to manually replace BUCKET_ID